### PR TITLE
@broskoski => Add banner to promoted content

### LINKED ISSUE #6: [Differentiation] Promoted banner for Articles
Mobile version of https://github.com/artsy/force-private/issues/5254

iphone 6 copy 4

Sketch file: https://www.dropbox.com/s/d4x3may28sdv2jr/DifferentiationMobileSpec.sketch?dl=0

### PR DESCRIPTION
**Pull request by eessex**
closes #6 

Add banner to articles in promoted content channels.

### CODE REVIEW

File: apps/articles/stylesheets/index.styl [hunk 1]
```diff
@@ -6,3 +6,26 @@
...
+  padding-top 0
+
+.articles-promoted
+  text-decoration none
+  display flex
+  justify-content space-between
+  margin 0px 20px
+  padding 15px 0
+  border-bottom 1px solid #ccc
+  &__title
```
Comment (broskoski) [inline apps/articles/stylesheets/index.styl:20]: Nice BEM-ing 👍

File: apps/articles/client/editorial_signup.coffee [hunk 2]
```diff
@@ -4,6 +4,7 @@ Backbone = require 'backbone'
 editorialSignupLushTemplate = -> require('../templates/editorial_signup_lush.jade') arguments...
 Cycle = require '../../../components/cycle/index.coffee'
 { resize } = require '../../../components/resizer/index.coffee'
+analyticsHooks = require '../../../lib/analytics_hooks.coffee'
```
Comment (broskoski) [inline apps/articles/client/editorial_signup.coffee]: Looks like this isn't used in this view, is this a mistype or something that is going to be addressed later?

File: apps/articles/client/article.coffee [hunk 3]
```diff
@@ -95,6 +101,27 @@ module.exports = class ArticleIndexView extends Backbone.View
...
+            success: (profile) ->
+              $('#article-body-container').addClass('promoted').prepend promotedTemplate
+                name: partner.get('name')
+                href: profile.href()
+    else if @article.get('channel_id') is sd.PC_AUCTION_CHANNEL
+      new Sale(id: @article.get('auction_ids')?[0]).fetch
+        error: -> @$el('.articles-promoted').hide()
+        success: (sale) ->
+          $('.article-section-header').hide()
+          $('#article-body-container').addClass('promoted').prepend promotedTemplate
```
Comment (broskoski) [inline apps/articles/client/article.coffee]: So I think there is an opportunity to refactor this to be a bit easier to read (and process).

In both cases, you're adding a class to the body-container and prepending a template. It probably makes sense to pull this out and make it its own method on the view. Something like:

``` coffeescript
renderPromotedTemplate: (name, href, type) ->
   $('#article-body-container').addClass('promoted').prepend promotedTemplate
      name: name
      href: href
      type: type
```

This makes it easier to test as well, you wouldn't have to "fake" the successful fetch. This isn't vital to address before merging, more just something to think about!

Also, it's good practice to scope any jQuery to the view (i.e. `@$('.article-section-header').hide()`). See http://backbonejs.org/#View-dollar

**Comment by broskoski** [comment]
Looks good to me, feel free to update with refactors or merge and address later!

File: apps/articles/test/client/article.coffee [hunk 4]
```diff
@@ -91,10 +102,36 @@ describe 'ArticleIndexView', ->
...
+      @options.sd.PC_ARTSY_CHANNEL = '5086df098523e60002000012'
+      @ArticleIndexView.__set__ 'sd', { PC_AUCTION_CHANNEL: '5086df098523e60002000011', PC_ARTSY_CHANNEL: '5086df098523e60002000012', ARTICLE: @model.attributes }
+      benv.render resolve(__dirname, '../../templates/article.jade'), @options, =>
+        @view = new @ArticleIndexView
+          el: $('body')
+        done()
+
+    it 'displays promoted content banner for auction', ->
+
+      Backbone.sync.args[5][2].success fabricate 'sale'
```
Comment (broskoski) [inline apps/articles/test/client/article.coffee:135]: Probably makes sense to write a quick test for the `error` case as well.

File: apps/articles/stylesheets/index.styl (hunk 1 above)
Comment (acjay) [reply apps/articles/stylesheets/index.styl:20]: <3 BEM

File: apps/articles/test/client/article.coffee (hunk 4 above)
Comment (eessex) [reply apps/articles/test/client/article.coffee:135]: In refactoring realized the error case was in mind of force, not relevant in microgravity -- ended up removing the error callbacks.

Comment (broskoski) [reply apps/articles/test/client/article.coffee:135]: 👍

**Comment by broskoski** [comment]
Looks good!

**Pull request merged**